utils/navigation_decorator.py

```diff
@@ -0,0 +1,12 @@
+import cfme.fixtures.pytest_selenium as sel
+import cfme.web_ui.menu as menu
+from method_decorator import method_decorator
+
+
+def navigate(page_name):
```

Comment:
This should eventually be build into the navigate system. @mfalesni we talked about this before, nav steps can have a fn that tries to determine if we are _on_ that nav page. Some of these will require context, in the case of "provider_edit" for example. Perhaps here is a good place where we can have a standardized id that tells us what page we are on. thoughts @mfalesni ??